web/core/store/notifications/workspace-notification.store.ts

```diff
@@ -0,0 +1,323 @@
+  // actions
+  setCurrentNotificationTab: (tab: TNotificationTab) => void;
+  getNotifications: (
+    workspaceSlug: string,
+    loader?: TNotificationLoader,
+    queryCursorType?: TNotificationQueryParamType
+  ) => Promise<TNotificationPaginatedInfo | undefined>;
+  getNotificationById: (workspaceId: string, notificationId: string) => Promise<TNotification | undefined>;
+  deleteNotificationById: (workspaceId: string, notificationId: string) => Promise<void>;
+  markAllNotificationsAsRead: (workspaceId: string) => Promise<void>;
+}
+
+export class WorkspaceNotificationStore implements IWorkspaceNotificationStore {
+  // constants
+  paginatedCount = 30;
+  // observables
+  loader: TNotificationLoader = undefined;
+  notifications: Record<string, INotification> = {};
+  currentNotificationTab: TNotificationTab = ENotificationTab.ALL;
+  paginationInfo: Omit<TNotificationPaginatedInfo, "results"> | undefined = undefined;
+  filters: TNotificationFilter = {
+    type: {
+      assigned: false,
+      created: false,
+      subscribed: false,
+    },
+    snoozed: false,
+    archived: false,
+    read: false,
+  };
+
+  constructor(private store: CoreRootStore) {
+    makeObservable(this, {
+      // observables
+      loader: observable.ref,
+      notifications: observable,
+      currentNotificationTab: observable.ref,
+      paginationInfo: observable,
+      filters: observable,
+      // computed
+      // helper actions
+      setCurrentNotificationTab: action,
+      mutateNotifications: action,
+      updateFilters: action,
+      updateBulkFilters: action,
+      // actions
+      getNotifications: action,
+      getNotificationById: action,
+      deleteNotificationById: action,
+      markAllNotificationsAsRead: action,
+    });
+  }
+
+  // computed
+
+  // computed functions
+  /**
+   * @description get notification ids by workspace id
+   * @param { string } workspaceId
+   */
+  notificationIdsByWorkspaceId = computedFn((workspaceId: string) => {
+    if (!workspaceId || isEmpty(this.notifications)) return undefined;
+    const workspaceNotificationIds = Object.values(this.notifications || {})
+      .filter((n) => n.workspace === workspaceId)
+      .filter((n) => {
+        if (!this.filters.archived && !this.filters.snoozed) {
+          if (n.archived_at) {
+            return false;
+          } else if (n.snoozed_till) {
+            return false;
+          } else {
+            return true;
+          }
+        } else {
+          if (this.filters.snoozed) {
+            return n.snoozed_till ? true : false;
+          } else if (this.filters.archived) {
+            return n.archived_at ? true : false;
+          } else {
+            return true;
+          }
+        }
+      })
+      // .filter((n) => (this.filters.read ? (n.read_at ? true : false) : n.read_at ? false : true))
+      .map((n) => n.id) as string[];
+    return workspaceNotificationIds ?? undefined;
```

Comment:
workspaceNotificationIds will atleast be an empty array. This check is not required.